src/helpers.js

```diff
@@ -192,45 +192,57 @@ export function log(context, status, ...message){
 
 /**
  * Get JSON options
- * @returns void
+ * ID priority: data-attribute > selector#id > unique id
+ * @version 0.0.1
+ * @returns {object}
  */
-export function getOptions(context){
+export function getOptions(context, defaultOptions){
+    const numeric = ['duration', 'activeSection']; // convert these props to float
+    const wrapper = context.wrapper;
+
     // options from attribute
-    let string = context.wrapper.getAttribute(context._attr.container);
+    let dataAttribute = wrapper.getAttribute(context._attr.container);
     let options = {};
 
-    if(!isJSON(string)){
-        context.id = context.options.id;
-        return;
+    // data attribute doesn't exist or not JSON format -> string
+    if(!dataAttribute || !isJSON(dataAttribute)){
+        // reassign id
+        const id = dataAttribute || wrapper.id || defaultOptions.id;
+        context.id = id;
+        defaultOptions.id = id;
+
+        return defaultOptions;
     }
 
-    // option priority: attribute > js object > default
-    options = {...JSON.parse(string)};
+    options = JSON.parse(dataAttribute);
 
-    // convert boolean string to real boolean
     for(const [key, value] of Object.entries(options)){
+        // convert boolean string to real boolean
         if(value === "false") options[key] = false;
-        if(value === "true") options[key] = true;
-        if(!isNaN(value)) options[key] = parseInt(value);
+        else if(value === "true") options[key] = true;
+        // convert string to float
+        else if(numeric.includes(key) && typeof value === 'string' && value.length > 0) options[key] = parseFloat(value);
+        else options[key] = value;
     }
 
-    // get ID
-    if(options['id'] && !isEmptyString(options['id'])){
-        context.id = options['id'];
-    }else{
-        context.id = context.options.id;
-    }
+    // reassign id
+    const id = options.id || wrapper.id || defaultOptions.id;
+    context.id = id;
+    options.id = id;
+
+    options = {...defaultOptions, ...options};
 
-    // replace default options
-    context.options = {...context.options, ...options};
+    // remove json
+    wrapper.removeAttribute(context._attr.container);
```

Comment:
Do not remove json string from attribute, please.

Reply:
I have removed it.